custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssueNavigator.kt

```diff
@@ -1,41 +1,66 @@
 import java.time.Duration
+import java.util.*
 
 class CloudIssueNavigator(
     driver: WebDriver
 ) {
+    private val webDriver = driver
     private val page = FalliblePage.Builder(
-            driver,
-            or(
-                and(
-                    or(
-                        presenceOfElementLocated(By.cssSelector("ol.issue-list")),
-                        presenceOfElementLocated(By.id("issuetable")),
-                        presenceOfElementLocated(By.id("issue-content"))
-                    ),
-                    or(
-                        presenceOfElementLocated(By.id("jira-issue-header")),
-                        presenceOfElementLocated(By.id("key-val"))
-                    ),
-                    or(
-                        presenceOfElementLocated(By.id("new-issue-body-container")),
-                        presenceOfElementLocated(By.className("issue-body-content"))
-                    )
+        driver,
+        or(
+            and(
+                or(
+                    presenceOfElementLocated(By.cssSelector("ol.issue-list")),
+                    presenceOfElementLocated(By.id("issuetable")),
+                    presenceOfElementLocated(By.id("issue-content"))
                 ),
-                presenceOfElementLocated(By.className("no-results-hint")) // TODO is it too optimistic like in SearchServerFilter.waitForIssueNavigator ?
-            )
+                or(
+                    presenceOfElementLocated(By.id("jira-issue-header")),
+                    presenceOfElementLocated(By.id("key-val"))
+                ),
+                or(
+                    presenceOfElementLocated(By.id("new-issue-body-container")),
+                    presenceOfElementLocated(By.className("issue-body-content"))
+                )
+            ),
+            presenceOfElementLocated(By.className("no-results-hint")) // TODO is it too optimistic like in SearchServerFilter.waitForIssueNavigator ?
         )
+    )
         .cloudErrors()
         .timeout(Duration.ofSeconds(30))
         .build()
 
     fun waitForNavigator() {
         page.waitForPageToLoad()
     }
+
+    fun chooseElement(): WebElement {
+        val elements = webDriver.wait(
+            ExpectedConditions.presenceOfAllElementsLocatedBy(By.xpath("//*[@class ='issue-list']/*"))
+        )
+        val rndIndex = Random().nextInt(elements.size - 1)
+        return elements[rndIndex]
+    }
+
+    fun switchBetweenIssuesInSearchResults(element: WebElement) {
+        val dataKey = element.getAttribute("data-key")
+        val title = element.getAttribute("title")
+        webDriver.wait(
+            ExpectedConditions.elementToBeClickable(By.xpath("//*[@data-key = '$dataKey']"))
+        )
+            .click()
+        webDriver.wait(
+            ExpectedConditions.presenceOfElementLocated(By.xpath("//*[@data-test-id = 'issue.views.issue-base.foundation.summary.heading' and contains(text(), '$title')]"))
```

Comment:
Please use `performance profile` to see which the last elements are being loaded and wait for them
as i could see it's `comments` section, `watching` button and `menu`